omnibus/package-scripts/ddot-rpm/preinst

```diff
@@ -0,0 +1,124 @@
+# The installation logic is handled by the installer in the following    #
+# file: pkg/fleet/installer/packages/datadog_agent_ddot_linux.go              #
+#                                                                        #
+##########################################################################
+
+INSTALL_DIR=/opt/datadog-agent
+
+set -e
+
+# Uninstall the agent if it was installed by the installer
+if command -v datadog-installer >/dev/null 2>&1 && datadog-installer is-installed datadog-agent-ddot; then
+    datadog-installer remove datadog-agent-ddot || printf "[ WARNING ]\tFailed to remove datadog-agent-ddot installed by the installer\n"
+fi
+
+# RPM Agents < 5.18.0 expect the preinst script of the _new_ package to stop the agent service on upgrade (which is defined with an init.d script on Agent 5)
+# So let's stop the Agent 5 service here until we don't want to support upgrades from Agents < 5.18.0 anymore
+if [ -f "/etc/init.d/datadog-agent-ddot" ]; then
+    /etc/init.d/datadog-agent-ddot stop || true
+fi
+
+# Run the upgrade prerm if supported in the previous version. See pkg/fleet/installer/packages/datadog_agent_ddot_linux.go
+if [ "$*" = "2" ] && [ -f "${INSTALL_DIR}/embedded/bin/installer" ] && ${INSTALL_DIR}/embedded/bin/installer is-prerm-supported; then
+    ${INSTALL_DIR}/embedded/bin/installer prerm --upgrade datadog-agent-ddot rpm || true
+    exit 0
+fi
+
+##########################################################################
+#    LEGACY INSTALLATION LOGIC BELOW WHEN UPGRADING FROM AGENT <7.68     #
+#                   DO NOT EDIT THIS SECTION                             #
+##########################################################################
+
+SERVICE_NAME=datadog-agent-ddot
+
+stop_agent() {
+    # Stop an already running agent
+    # Only supports systemd and upstart
+    if command -v systemctl >/dev/null 2>&1; then
+        systemctl stop $SERVICE_NAME || true
+    elif command -v initctl >/dev/null 2>&1; then
+        initctl stop $SERVICE_NAME || true
+    else
+        echo "[ WARNING ]\tCannot detect a supported init system. The datadog-agent-ddot package only provides service files for systemd and upstart."
+    fi
+}
+
+deregister_agent() {
+    # Disable agent start on system boot
+    # Only supports systemd and upstart
+    if command -v systemctl >/dev/null 2>&1; then
+        # Force systemd to ignore the sysvinit scripts. Only cosmetic, remove some irrelevant warnings during upgrade
+        SYSTEMCTL_SKIP_SYSV=true systemctl disable $SERVICE_NAME || true
+    elif command -v initctl >/dev/null 2>&1; then
+        # Nothing to do, this is defined directly in the upstart job file
+        :
+    else
+        echo "[ WARNING ]\tCannot detect a supported init system. The datadog-agent-ddot package only provides service files for systemd and upstart."
+    fi
+}
+
+stop_agent
+deregister_agent
+
+# RPM unpacks the new files before running prerm of the old package
```

Comment:
I'm unsure if the rest of this file is necessary for the DDOT preinst, this is copied from `agent-rpm/preinst`